src/gateio_new_coins_announcements_bot/new_listings_scraper.py

```diff
@@ -25,100 +24,17 @@
 previously_found_coins = set()
 
 
-def get_announcement():
-    """
-    Retrieves new coin listing announcements
-
-    """
-    logger.debug("Pulling announcement page")
-    # Generate random query/params to help prevent caching
-    rand_page_size = random.randint(1, 200)
-    letters = string.ascii_letters
-    random_string = "".join(random.choice(letters) for i in range(random.randint(10, 20)))
-    random_number = random.randint(1, 99999999999999999999)
-    queries = [
-        "type=1",
-        "catalogId=48",
-        "pageNo=1",
-        f"pageSize={str(rand_page_size)}",
-        f"rnd={str(time.time())}",
-        f"{random_string}={str(random_number)}",
-    ]
-    random.shuffle(queries)
-    logger.debug(f"Queries: {queries}")
-    request_url = (
-        f"https://www.binance.com/gateway-api/v1/public/cms/article/list/query"
-        f"?{queries[0]}&{queries[1]}&{queries[2]}&{queries[3]}&{queries[4]}&{queries[5]}"
-    )
-
-    latest_announcement = requests.get(request_url)
-    if latest_announcement.status_code == 200:
```

Comment:
The status code verification is missing in both *_scraper.py files.

Reply:
There are tests for this. The new scrapers use `response.raise_for_status()` to raise an HTTPError if the status is not 200

Reply:
The problem with raising an exception every time is that some errors can be ignored and should not crash the bot.
e.g. 429 Too Many Requests shouldn't cause an exception directly because the bot can continue running and after a few seconds/minutes it will work again/get valid responses.

Currently the develop branch ignores all responses that don't return status code 200 which might also not be the best approach.

I think logging the error and ignoring the announcement but keeping the bot running is more useful than crashing the bot.

Reply:
Oh wait, i just noticed that the whole `get_last_coi()` call is wrapped in an try...catch.